contributingGuides/review/RULES.md

```diff
@@ -0,0 +1,203 @@
+# ✅ AI Code Review Rules
+
+These rules are used to conduct structured code reviews on pull request diffs. Each rule includes:
+
+- A unique **Rule ID**
+- **Severity level**: MUST / SHOULD / OPTIONAL
+- **Pass/Fail condition**
+- Examples of good and bad usage
+
+---
+
+## Performance Rules
+
+### [PERF-1] No spread in list item's renderItem
+- **Severity**: MUST
+- **Condition**: When passing data to components in renderItem functions, avoid using spread operators to extend objects. Instead, pass the base object and additional properties as separate props to prevent unnecessary object creation on each render.
+
+Good:
```

Comment:
Would another good be?
```
item.isSelected = isSelected;
item.shouldAnimateInHighlight = isItemHighlighted;
<Component
  item={item}
/>
```

Reply:
Technically yes, but is definitely not idiomatic and property assignment at each render is just hard to track (no real component is this short) + error prone (could happen everywhere, inside functions). Sooner or later will swing back.

Reply:
Will this work as a piece of JS? Yes. Should we promote this as maintainable code? No.